### PR TITLE
ref: Migrate domain_record_info to InfoModule base class

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This pull request migrates the `linode.cloud.domain_record_info` module to the InfoModule base class. Additionally, this PR adds a new `InfoModuleParamGroup` type that allows mutually exclusive parameters to be defined.

## ✔️ How to Test

The following test steps assume you have pulled down this PR and run `make install`.

### Integration Testing

```
make test TEST_ARGS="-v domain_basic domain_list domain_record domain_zone_file"
```

### Manual Testing

1. In an ansible_linode sandbox environment (e.g. dx-devenv), run the following:

```yaml
- name: ansible_linode Sandbox Playbook
  hosts: localhost
  tasks:
    - name: Get information about the current user
      linode.cloud.profile_info: {}
      register: profile_info

    - name: Create a domain
      linode.cloud.domain:
        domain: "ansible-test-{{ profile_info.profile.username }}.com"
        type: master
        soa_email: "{{ profile_info.profile.email }}"
        state: present
      register: domain_create

    - name: Create a domain record
      linode.cloud.domain_record:
        domain_id: "{{ domain_create.domain.id }}"
        name: "test"
        type: A
        target: "8.8.8.8"
        state: present
      register: record_create

    - name: Get information about the record by domain name and record name
      linode.cloud.domain_record_info:
        domain: "{{ domain_create.domain.domain }}"
        name: "{{ record_create.record.name }}"

    - name: Get information about the record by domain ID and record ID
      linode.cloud.domain_record_info:
        domain_id: "{{ domain_create.domain.id }}"
        id: "{{ record_create.record.id }}"
```
2. Ensure the playbook runs successfully.
3. Ensure the output of the `Get information about the record by domain name and record name` task accurately reflects the new domain record.
4. Ensure the output of the `Get information about the record by domain ID and record ID` task accurately reflects the new domain record.

### CODE REVIEW

File: plugins/module_utils/linode_common_info.py [hunk 1]
```diff
@@ -41,6 +42,28 @@ class InfoModuleParam:
...
 
+class InfoModuleParamGroupPolicy(Enum):
+    """
+    Defines the policies that can be set for a param group.
+    """
+
+    EXACTLY_ONE_OF = "exactly_one_of"
+
+
+class InfoModuleParamGroup:
```
Comment (lgarber-akamai) [inline plugins/module_utils/linode_common_info.py:53]: This addition allows for required but mutually exclusive params (e.g. `domain` and `domain_id`)

File: plugins/module_utils/linode_common_info.py [hunk 2]
```diff
@@ -231,14 +274,22 @@ def run(self) -> None:
...
-        )
+        base_module_args = {
+            "module_arg_spec": self.module_arg_spec,
+            "required_one_of": [attribute_names],
+            "mutually_exclusive": [attribute_names],
+        }
+
+        if len(attribute_names) > 0:
+            base_module_args["required_one_of"].append(attribute_names)
+            base_module_args["mutually_exclusive"].append(attribute_names)
```
Comment (lgarber-akamai) [inline plugins/module_utils/linode_common_info.py:287]: Derived from the fix in #562

File: plugins/modules/domain_record_info.py [hunk 3]
```diff
@@ -1,193 +1,105 @@
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
 
-"""This module contains all of the functionality for Linode Domain records."""
+"""This module allows users to retrieve information about a Linode VPC Subnet."""
```
Comment (ezilber-akamai) [inline plugins/modules/domain_record_info.py:4]: I think this should say `Linode Domain record` instead of `Linode VPC Subnet`.

Comment (lgarber-akamai) [reply plugins/modules/domain_record_info.py:4]: oops, nice catch!

**Review by jriddle-linode — APPROVED** [review]
LGTM, good test steps, works locally.

**Review by ezilber-akamai — APPROVED** [review]
Integration and manual tests are working well locally. LGTM

**Review by zliang-akamai — APPROVED** [review]

**Pull request merged**